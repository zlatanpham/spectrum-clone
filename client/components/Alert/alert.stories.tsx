import { storiesOf } from '@storybook/react';
import React from 'react';
import Alert from '.';

storiesOf('components/Alert', module).add('Basic', () => (
  <div className="p-5">
    <Alert type="warning" className="mb-3" style={{ width: '500px' }}>
      <strong className="text-black">
        Trying to mute a specific conversation?
      </strong>{' '}
      You can turn off email notifications for individual threads by clicking on
      the notification icon
    </Alert>
    <Alert type="error" className="mb-3" style={{ width: '500px' }}>
      <strong className="text-black">
        Trying to mute a specific conversation?
      </strong>{' '}
      You can turn off email notifications for individual threads by clicking on
      the notification icon
    </Alert>
    <Alert type="success" className="mb-3" style={{ width: '500px' }}>
      <strong className="text-black">
        Trying to mute a specific conversation?
      </strong>{' '}
      You can turn off email notifications for individual threads by clicking on
      the notification icon
    </Alert>
    <Alert type="info" style={{ width: '500px' }}>
      <strong className="text-black">
        Trying to mute a specific conversation?
      </strong>{' '}
      You can turn off email notifications for individual threads by clicking on
      the notification icon
    </Alert>
  </div>
));
