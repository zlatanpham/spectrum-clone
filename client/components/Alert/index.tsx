import React from 'react';
import styled from 'styled-components';
import tw from 'tailwind.macro';

const Container = styled.div`
  ${tw`rounded border px-3 py-2 text-sm`}
`;

const ErrorAlert = styled(Container)`
  ${tw`border-red bg-pink-100 text-red-500`}
`;

const WarningAlert = styled(Container)`
  ${tw`border-yellow-400 bg-yellow-200 text-brown-100`}
`;

const SuccessAlert = styled(Container)`
  ${tw`border-green bg-green-100 text-green-500`}
`;

const InfoAlert = styled(Container)`
  ${tw`border-blue bg-blue-200 text-blue-dark`}
`;

export type AlertType = 'error' | 'warning' | 'info' | 'success';

export interface AlertProps {
  type: AlertType;
  className?: string;
  style?: Object;
}

const components: { [k in AlertType]: React.ElementType } = {
  error: ErrorAlert,
  success: SuccessAlert,
  warning: WarningAlert,
  info: InfoAlert,
};

const Alert: React.FC<AlertProps> = props => {
  const { type = 'warning', children, ...rest } = props;
  const RenderAlert = components[type];
  return <RenderAlert {...rest}>{children}</RenderAlert>;
};

export default Alert;
