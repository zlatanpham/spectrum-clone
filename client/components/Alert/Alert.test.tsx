import React from 'react';
import { render } from '@testing-library/react';
import Alert from '.';

describe('Alert', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Alert type="error" className="mb-3" style={{ width: '500px' }}>
        <strong className="text-black">
          Trying to mute a specific conversation?
        </strong>{' '}
        You can turn off email notifications for individual threads by clicking
        on the notification icon
      </Alert>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
