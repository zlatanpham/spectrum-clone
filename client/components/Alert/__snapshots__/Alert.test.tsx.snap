// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot 1`] = `
.c0 {
  border-radius: 0.25rem;
  border-width: 1px;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.875rem;
  border-color: #FF3B30;
  background-color: #FFEDF6;
  color: #85000C;
}

<div
  class="c0 mb-3"
  style="width: 500px;"
>
  <strong
    class="text-black"
  >
    Trying to mute a specific conversation?
  </strong>
   
  You can turn off email notifications for individual threads by clicking on the notification icon
</div>
`;
